pkg/kubelet/cm/memorymanager/memory_manager_test.go

```diff
@@ -213,3 +298,2112 @@ func TestConvertPreReserved(t *testing.T) {
+				t.Fatalf("The actual machine state: %v is different from the expected one: %v", mgr.state.GetMachineState(), testCase.expectedMachineState)
+			}
+		})
+	}
+}
+
+func TestNewManager(t *testing.T) {
+	machineInfo := cadvisorapi.MachineInfo{
+		Topology: []cadvisorapi.Node{
+			{
+				Id:     0,
+				Memory: 10 * gb,
+				HugePages: []cadvisorapi.HugePagesInfo{
+					{
+						PageSize: pageSize1Gb,
+						NumPages: 5,
+					},
+				},
+			},
+			{
+				Id:     1,
+				Memory: 10 * gb,
+				HugePages: []cadvisorapi.HugePagesInfo{
+					{
+						PageSize: pageSize1Gb,
+						NumPages: 5,
+					},
+				},
+			},
+		},
+	}
+	expectedReserved := systemReservedMemory{
+		0: map[v1.ResourceName]uint64{
+			v1.ResourceMemory: 1 * gb,
+		},
+		1: map[v1.ResourceName]uint64{
+			v1.ResourceMemory: 1 * gb,
+		},
+	}
+	testCases := []testMemoryManager{
+		{
+			description:                "Successfuly created Memory Manager instance",
+			policyName:                 "single-numa",
+			machineInfo:                machineInfo,
+			nodeAllocatableReservation: v1.ResourceList{v1.ResourceMemory: *resource.NewQuantity(2*gb, resource.BinarySI)},
+			preReservedMemory: map[int]map[v1.ResourceName]resource.Quantity{
+				0: nodeResources{v1.ResourceMemory: *resource.NewQuantity(gb, resource.BinarySI)},
+				1: nodeResources{v1.ResourceMemory: *resource.NewQuantity(gb, resource.BinarySI)},
+			},
+			affinity: topologymanager.NewFakeManager(),
+			expectedError:   nil,
+			expectedReserved: expectedReserved,
+		},
+		{
+			description:                "Should return an error where preReservedMemory is not correct",
+			policyName:                 "single-numa",
+			machineInfo:                machineInfo,
+			nodeAllocatableReservation: v1.ResourceList{v1.ResourceMemory: *resource.NewQuantity(2*gb, resource.BinarySI)},
+			preReservedMemory: map[int]map[v1.ResourceName]resource.Quantity{
+				0: nodeResources{v1.ResourceMemory: *resource.NewQuantity(gb, resource.BinarySI)},
+				1: nodeResources{v1.ResourceMemory: *resource.NewQuantity(2*gb, resource.BinarySI)},
+			},
+			affinity: topologymanager.NewFakeManager(),
+			expectedError:   fmt.Errorf("the total amount of memory of type \"memory\" is not equal to the value determined by Node Allocatable feature"),
+			expectedReserved: expectedReserved,
+		},
+		{
+			description:                "Should return an error when memory reserved for system is empty (preReservedMemory)",
+			policyName:                 "single-numa",
+			machineInfo:                machineInfo,
+			nodeAllocatableReservation: v1.ResourceList{},
+			preReservedMemory:          map[int]map[v1.ResourceName]resource.Quantity{},
+			affinity:                   topologymanager.NewFakeManager(),
+			expectedError:                     fmt.Errorf("[memorymanager] you should specify the memory reserved for the system"),
```

Comment:
`the memory reserved for the system` -> `system reserved memory` or `the amount of memory reserved for the system`?

Reply:
Your propositions look nicer, done.